core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -70,6 +70,13 @@ private[ui] class RDDOperationCluster(val id: String, private var _name: String)
   def getAllNodes: Seq[RDDOperationNode] = {
     _childNodes ++ _childClusters.flatMap(_.childNodes)
   }
+
+  /** Return all the nodes which are cached. */
+  def getCachedNodes: Seq[RDDOperationNode] = {
+    val cachedNodes = _childNodes.filter(_.cached)
+    _childClusters.foreach(cluster => cachedNodes ++= cluster._childNodes.filter(_.cached))
```

Comment:
also, another way to rewrite this would be:
```
_childNodes.filter(_.cached) ++ _childClusters.flatMap(_.getCachedNodes)
```
I think it's both more concise and easier to read